pinot-core/src/main/java/com/linkedin/pinot/core/segment/index/readers/StringDictionary.java

```diff
@@ -55,14 +55,13 @@ public String get(int dictionaryId) {
     if ((dictionaryId == -1) || (dictionaryId >= length())) {
       return "null";
     }
-    String val = getString(dictionaryId);
-    byte[] bytes = val.getBytes(UTF_8);
-    for (int i = 0; i < lengthofMaxEntry; i++) {
-      if (bytes[i] == paddingChar) {
-        return new String(bytes, 0, i, UTF_8);
+    byte[] bytes = dataFileReader.getBytes(dictionaryId, 0);
+    for (int i = lengthofMaxEntry - 1; i >= 0; i--) {
+      if (bytes[i] != paddingChar) {
```

Comment:
one note based on our discussion. This change is neither guaranteed to be efficient. It will also trailing padding characters as part of strings. So, we have to guide customers to avoid having padding characters as part of string